warehouse/query-core/src/test/java/datawave/query/planner/rules/NodeTransformVisitorTest.java

```diff
@@ -65,18 +66,37 @@ private void testSimplify(String query, String expected) throws Exception {
         testPushdown(query, expected, Collections.singletonList(regexSimplifier));
     }
     
-    private void testPushdown(String query, String expected, List<NodeTransformRule> rules) throws Exception {
+    private void testPushdown(String original, String expected, List<NodeTransformRule> rules) throws Exception {
         // create a query tree
-        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        ASTJexlScript originalScript = JexlASTHelper.parseJexlQuery(original);
         
         MockMetadataHelper helper = new MockMetadataHelper();
         
         // apply the visitor
-        script = NodeTransformVisitor.transform(script, rules, new ShardQueryConfiguration(), helper);
+        ASTJexlScript resultScript = NodeTransformVisitor.transform(originalScript, rules, new ShardQueryConfiguration(), helper);
         
-        // test the query tree
-        String result = JexlStringBuildingVisitor.buildQuery(script);
-        assertEquals("Unexpected transform", expected, result);
+        // Verify the script is as expected, and has a valid lineage.
+        assertScriptEquality(resultScript, expected);
+        assertLineage(resultScript);
+        
+        // Verify the original script was not modified, and still has a valid lineage.
+        assertScriptEquality(originalScript, original);
+        assertLineage(originalScript);
+        
+    }
+    
+    private void assertScriptEquality(ASTJexlScript actualScript, String expected) throws ParseException {
+        ASTJexlScript expectedScript = JexlASTHelper.parseJexlQuery(expected);
+        TreeEqualityVisitor.Comparison comparison = TreeEqualityVisitor.checkEquality(expectedScript, actualScript);
+        if (!comparison.isEqual()) {
+            log.error("Expected " + PrintingVisitor.formattedQueryString(expectedScript));
```

Comment:
I almost feel like I'd rather see this in as the unit test failure reason instead of `comparison.getReason()`. I'm not too in-tune with the query code but it looks like the `comparison.getReason()` will be the local difference. Is that more helpful for debugging/fixing than the Expected/Actual message here?